nvflare/apis/impl/controller.py

```diff
@@ -92,12 +92,17 @@ def __init__(self, task_check_period=0.2):
         self._all_done = False
         self._task_lock = Lock()
         self._task_monitor = threading.Thread(target=self._monitor_tasks, args=())
-        self._task_check_period = task_check_period
         self._dead_client_reports = {}  # clients that reported the job is dead on it: name => report time
         self._dead_clients_lock = Lock()  # need lock since dead_clients can be modified from different threads
         # make sure _check_tasks, process_task_request, process_submission does not interfere with each other
         self._controller_lock = Lock()
 
+        if not isinstance(task_check_period, (int, float)):
```

Comment:
@yanchengnv I noticed this check was done in ScatterAndGather and ModelController rather than in the base class. Should it be better done here?

Reply:
Yes. We won't touch ScatterAndGather for now, just to keep scope of changes minimal. validation_utils was added later and we didn't get to change old places.

Reply:
Okay, I put this check back into ModelController